src/conditionalLogicService/conditionallyShowOption.ts

```diff
@@ -200,13 +211,35 @@ export default function conditionallyShowOption(
       )
     },
   )
+  if (!validPredicates.length) return 'SHOW'
 
-  if (!validPredicates.length) return true
-  return validPredicates.some((predicate) =>
-    conditionallyShowOptionByPredicate(
+  let isLoadingDynamicDEPENDENCY = false
+  let isLoadingStaticDEPENDENCY = false
+  for (const predicate of validPredicates) {
+    const predicateResult = conditionallyShowOptionByPredicate(
       formElementsCtrl,
       predicate,
       optionsEvaluated,
-    ),
-  )
+    )
+    switch (predicateResult) {
+      case 'SHOW': {
+        return 'SHOW'
+      }
+      case 'LOADING_DYNAMIC_DEPENDENCY': {
+        isLoadingDynamicDEPENDENCY = true
+        break
+      }
+      case 'LOADING_STATIC_DEPENDENCY': {
+        isLoadingStaticDEPENDENCY = true
+        break
```

Comment:
```suggestion
        return 'SHOW'
```

This was the previous logic

Reply:
nope, i missed this being in the next function up.